packages/core/src/percy.js

```diff
@@ -235,6 +236,11 @@ export default class Percy {
       throw new Error('Not running');
     }
 
+    if (!this.envWarning && (!options.clientInfo || !options.environmentInfo)) {
+      this.envWarning = true;
+      this.log.warn('Warning: Missing `clientInfo` and/or `environmentInfo` properties');
+    }
+
```

Comment:
Thinking about this and seeing other changes, it doesn't quite feel like the right place to warn.

For example, some SDKs iterate over a list and use the `snapshot` method directly. They correctly provide env and client info, but they don't provide it with snapshots. So this warning would incorrectly trigger. 

As another example, the tests that you've changed previously had valid use cases where you can use the snapshot method without client/env info. If that info were to be required for every snapshot, this whole PR would be made invalid by adding the properties to the validation schema as required properties.

Now that I'm noticing this warning, I believe my previous comment about the env info being in the wrong place in the snapshot command was due to this warning. In that command, the env/client info should be passed on init and not with every snapshot. This also explains why you needed to duplicate the client info that was already present.

A better place for this would be in client. Although when discussing this offline, we discovered it was too eager to do so at the construction of the UA header because we didn't want to trigger the warning when a build is created. I think we're on the right path moving it to the same place snapshots are handled, but instead, we should move it to the `client.createSnapshot` method. That way we avoid the check for builds, but still check at the last moment before it should be there.